pages/index.tsx

```diff
@@ -5,163 +5,109 @@ import Drawer from "@/components/UI/Drawer/Drawer";
+import { locationsURL } from "@/utils/urls";
 import HeadWithMeta from "@/components/base/HeadWithMeta/HeadWithMeta";
 import FilterMenu from "@/components/UI/FilterMenu/FilterMenu";
 import { serverSideTranslations } from "next-i18next/serverSideTranslations";
 import { useTranslation, Trans } from "next-i18next";
-import {
-  initialReasoningFilter,
-  ReasoningFilterMenuOption,
-} from "@/components/UI/FilterMenu/FilterReasoningMenu";
 import { useRouter } from "next/router";
 import LocaleSwitch from "@/components/UI/I18n/LocaleSwitch";
+import { useURLActions, useURLStore } from "@/stores/urlStore";
 
 const LeafletMap = dynamic(() => import("@/components/UI/Map"), {
   ssr: false,
 });
 
-const getReasoningFilter = (
-  reasoningFilterMenuOption: ReasoningFilterMenuOption
-) => {
-  reasoningFilterMenuOption.type;
-  if (reasoningFilterMenuOption.type === "channel") {
-    return undefined;
-  }
-
-  if (reasoningFilterMenuOption.type === "reason") {
-    return reasoningFilterMenuOption.value;
-  }
-};
-
 type Props = {
   deviceType: DeviceType;
   singleItemDetail: any;
 };
 export default function Home({ deviceType, singleItemDetail }: Props) {
   const { t } = useTranslation(["common", "home"]);
   const router = useRouter();
-  const [slowLoading, setSlowLoading] = useState(false);
-  const [reasoningFilterMenuOption, setReasoningFilterMenuOption] =
-    useState<ReasoningFilterMenuOption>(initialReasoningFilter);
+  const [sendRequest, setSendRequest] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
   const [newerThanTimestamp, setNewerThanTimestamp] = useState<
     number | undefined
   >(undefined);
-  const [url, setUrl] = useState<string | null>(null);
   const device = useDevice();
   const isMobile = device === "mobile";
+  const { eventType } = useMapStore();
+  const { url } = useURLStore();
+  const { setReasoningFilterMenuOption } = useURLActions();
 
-  const coordinatesAndEventType:
-    | CoordinatesURLParametersWithEventType
-    | undefined = useCoordinates();
-
-  const urlParams = useMemo(() => {
-    const reasoningFilterValue = getReasoningFilter(reasoningFilterMenuOption);
-    return new URLSearchParams({
-      ne_lat: coordinatesAndEventType?.ne_lat,
-      ne_lng: coordinatesAndEventType?.ne_lng,
-      sw_lat: coordinatesAndEventType?.sw_lat,
-      sw_lng: coordinatesAndEventType?.sw_lng,
-      time_stamp: newerThanTimestamp ? newerThanTimestamp : undefined,
-      ...(reasoningFilterValue ? { reason: reasoningFilterValue } : {}),
-    } as any).toString();
-  }, [
-    coordinatesAndEventType?.ne_lat,
-    coordinatesAndEventType?.ne_lng,
-    coordinatesAndEventType?.sw_lat,
-    coordinatesAndEventType?.sw_lng,
-    newerThanTimestamp,
-    reasoningFilterMenuOption,
-  ]);
+  const { setDevice } = useMapActions();
 
-  const { error, isLoading, isValidating } = useSWR<DataLite | undefined>(
```

Comment:
swri neden kaldirdin? swr tek basina datayi guncel tutmak icin birkac ozellik iceriyor. refetchOnWindowFocus gibi. emin miyiz? eger eminsek ve baska bir yerde kullanilmiyorsa dependenciesden de ucurulmali.

Reply:
swr'i `locationsURL` icin atmaya devam edeceğiz çünkü `${BASE_URL}/feeds/${locationId}` gibi bir istegi cacheleyebiliriz fakat areas icin attigimiz request'te kullandigimiz `lat` `lng` degerlerinin precisioni cok yuksek ve ayni istegi iki kere atmamız cok düşük bir ihtimal. Ayrıca areas requestinin her zaman güncel olmasını istiyoruz.

swr'in dokumantasyonundan `refetchOnWindowFocus`'i okuyacağım. Dediğin gibi kullanmamız gereken bir seyse swr'a geri çekebilirim.